Apple & Google Sign In

cdanwards opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ [#7079](https://github.com/Expensify/App/issues/7079)

NOTE: Reading through `contributingGuides/APPLE_GOOGLE_SIGNIN.md` provides more details about testing in development as well as rationale for design decisions.

## Tests

### Apple Sign In

#### iOS
- [ ] Ensure app builds correctly
- [ ] Select the Apple Icon Button
- [ ] Enter valid Apple ID and Password
- [ ] If prompted, validate with 2 factor authentication
- [ ] Select to share email with app
- [ ] Select continue and sign in with Expensify
- [ ] Ensure logged in correctly

#### Android
- [ ] Ensure app builds correctly
- [ ] Select the Apple Icon Button
- [ ] Enter valid Apple ID and Password
- [ ] If prompted, validate with 2 factor authentication
- [ ] Select to share email with app
- [ ] Select continue and sign in with Expensify
- [ ] Ensure logged in correctly

#### Web
- [ ] Ensure no errors in console
- [ ] Select the Apple Icon Button
- [ ] Ensure Pop Up window appears
- [ ] Enter valid Apple ID and Password
- [ ] If prompted, validate with 2 factor authentication
- [ ] Select to share email with app, if prompted.
- [ ] Select continue and sign in with Expensify
- [ ] Ensure correct redirect to Expensify tab
- [ ] Ensure login occurs

#### Desktop
- [ ] Ensure no errors in console
- [ ] Select the Apple Icon Button
- [ ] Check that is redirects to web tab, url will be `new.expensify.com/sign-in-with-apple` (It will automatically redirect to home login because of redirects)
- [ ] Navigate to `staging.new.expensify.com/sign-in-with-apple`
- [ ] Ensure page loads and there is a `Continue with Apple` button.
- [ ] Select Button
- [ ] Go through web login flow
- [ ] After successful login, ensure there is a prompt to go back to desktop app
- [ ] Select option to navigate back to desktop app
- [ ] Ensure successful navigation
- [ ] Check for successful login once navigated back to desktop app



### Google Sign In

#### iOS
- [ ] Ensure app builds correctly
- [ ] Select the Google Icon Button
- [ ] Enter valid email and Password
- [ ] If prompted, validate with 2 factor authentication
- [ ] Ensure logged in correctly

#### Android
- [ ] Ensure app builds correctly
- [ ] Select the Google Icon Button
- [ ] Enter valid email and Password
- [ ] If prompted, validate with 2 factor authentication
- [ ] Select to share email with app
- [ ] Select continue and sign in with Expensify
- [ ] Ensure logged in correctly

#### Web
- [ ] Ensure no errors in console
- [ ] Select the Google Icon Button
- [ ] Ensure Pop Up window appears
- [ ] Enter valid email and Password
- [ ] If prompted, validate with 2 factor authentication
- [ ] Select to share email with app, if prompted.
- [ ] Ensure correct redirect to Expensify tab
- [ ] Ensure login occurs

#### Desktop
- [ ] Ensure no errors in console
- [ ] Select the Google Icon Button
- [ ] Check that is redirects to web tab, url will be `new.expensify.com/sign-in-with-google` (It will automatically redirect to home login because of redirects)
- [ ] Navigate to `staging.new.expensify.com/sign-in-with-google`
- [ ] Ensure page loads and there is a `Continue with Google` button.
- [ ] Select Button
- [ ] Go through web login flow
- [ ] After successful login, ensure there is a prompt to go back to desktop app
- [ ] Select option to navigate back to desktop app
- [ ] Ensure successful navigation
- [ ] Check for successful login once navigated back to desktop app

### Offline tests

Feature is not available in offline mode

### QA Steps

Same as Tests

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.


### Screenshots/Videos
<details>
<summary>Web</summary>

Apple:

https://github.com/Expensify/App/assets/8878532/0328c51e-2fdc-4fa5-be0c-cee694464b95

Google:

https://github.com/Expensify/App/assets/8878532/03547193-3f11-4f51-b32a-2955351e3a01

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/8878532/bd397966-44e8-4524-98ca-b5d58f3c9c73


</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/8878532/2440facb-af7b-4bdd-9fb5-8adf8131ceb0


</details>

<details>
<summary>Desktop</summary>

Apple:

https://github.com/Expensify/App/assets/8878532/80bd9e51-5835-4c97-b5e7-65646e929789

Google:

https://github.com/Expensify/App/assets/8878532/06218640-3b3b-4be6-bde9-11525d1f3192

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/8878532/66fd66f3-7518-4da0-a59d-53551b308dc6

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/8878532/8e3da5f5-710a-4b4f-ac45-db124ce7a80d

</details>

cdanwards committed:
Create DesktopRedirectPage

cdanwards committed:
Added Desktop Login method to onyx

cdanwards committed:
Set up everything to display redirect screen

cdanwards committed:
Switch to using DeeplinkRedirectLoadingIndicator

cdanwards committed:
Redirection to desktop flow is working

cdanwards committed:
Revert back Deeplinkredirectloadingindicator

cdanwards committed:
Updated logic on DesktopRedirectPage

cdanwards committed:
Fix navigation logic when navigating back to Desktop

cdanwards committed:
Add navigation to HOME for continue to web app

cdanwards committed:
Cleanup after self review

cdanwards committed:
Fix issue with google sign in button

cdanwards committed:
Switch from onyx key to url pathname redirect approach

cdanwards committed:
Remove console log

cdanwards committed:
Added Spanish Translations

cdanwards committed:
Removed unused line

cdanwards committed:
Some updates after self-review

cdanwards committed:
Reverting a change

cdanwards committed:
Adds loadingindicator when making expensify api call

cdanwards committed:
fixed prop

cdanwards committed:
Added react-native-google-signin

cdanwards committed:
Added contributing guide text

cdanwards committed:
Ran pod install

cdanwards committed:
Added url scheme to info plist

cdanwards committed:
Add google ids to const

cdanwards committed:
Adds Desktop Pages

cdanwards committed:
Adds jest config fix

cdanwards committed:
Adds linkingConfig and Icon

cdanwards committed:
Adds Google Sign In Buttons

cdanwards committed:
Adds begin google sign in function

cdanwards committed:
Updates LoginForm ThirdPartySignInPage and styles

lindboe committed:
Fix staging configuration

lindboe committed:
New fix: use shared values in CONFIG

lindboe committed:
Remove unused const values

cdanwards committed:
Fix build error

lindboe committed:
Fix typos

lindboe committed:
Fix Google button width

lindboe committed:
Simplify Google Sign-In button

lindboe committed:
Revert "Merge pull request #23295 from Expensify/revert-16723-apple-sign-in"

This reverts commit 948f7a1d2df36527e6ec34408320a90c9fd46a77, reversing
changes made to 99519f2347f91a2328f0e143e8a2859628412ffa.

cdanwards committed:
Rework to use DeeplinkRedirectLoadingIndicator

cdanwards committed:
Removed unused compose

cdanwards committed:
Fixes tests

cdanwards committed:
Moved DeeplinkWrapper to include PublicScreens - Checks to see if route should show deeplink prompt

cdanwards committed:
Remove console logs

cdanwards committed:
Prevent voice over for sign in with - redundant with apple and google buttons

cdanwards committed:
Fixed merge conflicts

cdanwards committed:
Merged main

lindboe committed:
Update contributing guide

Give more context on how the feature works, and update for free ngrok
tier changes

cdanwards committed:
Updated logic to check for authentication status as well as some additional criteria

cdanwards committed:
Removed comments

cdanwards committed:
Added comment to file

cdanwards committed:
Removed config setter

cdanwards committed:
Simplified if statements

cdanwards committed:
Merge branch 'Expensify:main' into new-third-party-signin

cdanwards committed:
Removed comment

cdanwards committed:
Fixes error on desktop google signin page and updates some variables and navigation methods

lindboe committed:
Merge pull request #48 from infinitered/lindboe/contributing-guide-update

New draft explaining how the feature works, and updated for free ngro…

cdanwards committed:
Changes Navigation function to grab screenName off of the state event

cdanwards committed:
Merge pull request #46 from infinitered/use-redirect-page

Rework to use DeeplinkRedirectLoadingIndicator

cdanwards committed:
Fixed merge conflicts from main

marcochavezf commented:
Assigning @0xmiroslav as C+ reviewer

OSBotify commented:
Use the links below to test this build in android and iOS. Happy testing!

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/23673/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/23673/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/23673/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/23673/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/23673/NewExpensify.dmg  | https://23673.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/23673/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://23673.pr-testing.expensify.com)  |

marcochavezf reviewed (changes requested):
Just a few minor style changes (mainly removing empty spaces) to keep consistency with the code base

cdanwards committed:
Update src/components/DeeplinkWrapper/index.website.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/AppleAuthWrapper/index.ios.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/AppleSignIn/index.android.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/AppleSignIn/index.android.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/AppleSignIn/index.android.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/AppleSignIn/index.desktop.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/AppleSignIn/index.ios.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/AppleSignIn/index.ios.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/AppleSignIn/index.website.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/GoogleSignIn/index.desktop.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/GoogleSignIn/index.native.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/GoogleSignIn/index.native.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/components/SignInButtons/IconButton.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/libs/Navigation/Navigation.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/libs/Navigation/shouldPreventDeeplinkPrompt/index.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/libs/actions/Session/index.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Update src/libs/actions/Session/index.js

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

cdanwards committed:
Remove extra line

cdanwards commented:
@marcochavezf thanks for all those suggestions. Just for future reference, is that style rule documented? I couldn't seem to find it in the contributing guide.

cdanwards committed:
Fix lint errors for JSDoc

marcochavezf commented:
> @marcochavezf thanks for all those suggestions. Just for future reference, is that style rule documented? I couldn't seem to find it in the contributing guide.

Ah good question, I think we haven't documented that rule, but I think it could be a good idea to document it because from the last commit seems the eslinter didn't catch errors in the JSDoc if there's a space between the comment and the function

cdanwards committed:
Merge branch 'main' into new-third-party-signin

situchan reviewed (commented):
Bug1: [web] Google logo background


<img width="280" alt="bug1" src="https://github.com/Expensify/App/assets/108292595/60578220-9fbc-4bd5-89f4-c39c81459bdc">

Bug2: [resized web] small top margin difference on mobile view (desktop view is perfect)


https://github.com/Expensify/App/assets/108292595/d697ab94-dc03-4194-b89c-9fbc7472ef13

Bug3: [web] offline behavior - nothing happens after 3rd party sign in success when disable network in chrome

https://github.com/Expensify/App/assets/108292595/1ec7cba2-d02b-42ea-a12a-57d55282e8b7


when disable network in system:

https://github.com/Expensify/App/assets/108292595/9b9e139d-d759-4e23-b283-4b903cd934a4



Bug4: [web] console error when start google signin process


<img width="606" alt="bug4" src="https://github.com/Expensify/App/assets/108292595/37d55130-d69c-471f-ae44-146014d010f2">

Bug5: [web] deeplink prompt not showing on signin page


https://github.com/Expensify/App/assets/108292595/105f29bc-b63f-459d-b744-a77cdcce587b

Bug 6: [web] Might be same root cause as Bug 1. white background


https://github.com/Expensify/App/assets/108292595/2b9318d7-02d9-4bc1-ae31-2fa9ce7190a6

Bug7: [web] apple logo font size is too large, compared to google

https://github.com/Expensify/App/assets/108292595/46c711a2-e0ef-4e13-9972-184458b04c1c


Bug8: [web] console error

<img width="1787" alt="bug8" src="https://github.com/Expensify/App/assets/108292595/eccb1e6e-608f-46e0-a0b5-73e6a75371a9">

Bug9: [desktop] infinite loading on google signin
<img width="1181" alt="bug9" src="https://github.com/Expensify/App/assets/108292595/88370aa5-1a4b-4b06-94a0-563d14c0fc6a">

https://github.com/Expensify/App/assets/108292595/ef66f771-1775-45bb-a229-c3945a5f9275

Bug10: [web/desktop] apple signin not working locally. I think this is known already. Should this be tested after deploy to staging?

Bug11: [ios] console error when apple signin first time on iOS device

![bug11](https://github.com/Expensify/App/assets/108292595/87b67996-ef94-471e-9ac3-59f5b43b89e9)

cdanwards commented on src/CONST.js:
```diff
@@ -481,6 +482,18 @@ const CONST = {
 
+    SIGN_IN_FORM_WIDTH: 300,
+
+    GOOGLE_SIGN_IN_WEB_CLIENT_ID: '921154746561-gpsoaqgqfuqrfsjdf8l7vohfkfj7b9up.apps.googleusercontent.com',
+    GOOGLE_SIGN_IN_IOS_CLIENT_ID: '921154746561-s3uqn2oe4m85tufi6mqflbfbuajrm2i3.apps.googleusercontent.com',
```
This is a good point! We'll update to both be in `CONFIG`

cdanwards committed:
Added required props and revert style change on mobile web

cdanwards committed:
Move google config to the CONFIG file

cdanwards committed:
Reverte another style change

cdanwards commented:
@situchan thanks for the thorough review! I'm leaving responses here to your comments:

Bug1: This is presented by the iframe used by google. This does not appear in non-development environments. More documentation is in the [Apple & Google Contributing Guide](https://github.com/Expensify/App/pull/23673/files#diff-65eb6935ab3443da2da663ae81ced5434837b39df30f03185c3297866214b34e)
Bug2: This was a regression based on the SignInPage wrapping its contents in justifyContent center. I'm pinging !@marco to clarify this.
Bug3: This feature shouldn't be available in offline mode. Going to continue to investigate how that call is being made.
Bug4: Good catch. Continuing to investigate.
Bug5: What browser are you using? Firefox isn't supported, and limited support for Safari, I believe. Have you tried a clean install of node modules? Is there an error in the console? You also might want to double check that you're config environment is set to something other than DEV. More documentation is in the [Apple & Google Contributing Guide](https://github.com/Expensify/App/pull/23673/files#diff-65eb6935ab3443da2da663ae81ced5434837b39df30f03185c3297866214b34e).
Bug6: Same as Bug1
Bug7: Because we're using the Apple & Google API's to generate the buttons, we don't have full control over this. Apple does let us customize the button more than Google, but font size isn't in the official list of adjustable props: https://appleid.apple.com/signinwithapple/button. We can ask design to sign off on it.
Bug8: Good catch. We've updated that component to provide the required props and eliminate that console error.
Bug9: Do you mind checking what network calls are being made? I'm unable to reproduce. (This would be an issue with the Expensify API if there is one)
Bug10: Correct that Apple on web is not able to be tested on localhost. There are instructions in the [Apple & Google Contributing Guide](https://github.com/Expensify/App/pull/23673/files#diff-65eb6935ab3443da2da663ae81ced5434837b39df30f03185c3297866214b34e) to set up a SSH tunnel locally to test this functionality.
Bug11: Continuing to investigate, but seems to be related to attempting to sign in when the user is not signed in with their Apple ID on their device.

TL;DR: Investigation continuing on bugs 3, 4, and 11

cdanwards commented:
Also will be fixing the merge conflicts.

cdanwards committed:
Change minHeight on TopSpacer to eliminate visual regression

cdanwards committed:
Merge main

cdanwards committed:
Removed unused import

marcochavezf commented:
> Bug2: This was a regression based on the SignInPage wrapping its contents in justifyContent center. I'm pinging !@marco to clarify this.

Just for visibility, this was discussed [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1691615406318989)

cdanwards commented:
@situchan some follow up on the rest of the bugs:

Bug3: This looks to be just a result of browser behavior. This is an artifact of using "Offline" simulation in the browser. It affects the window you're in, but not the pop-up window. If you go fully offline, like turn your wifi off, the pop-up doesn't render anything, which is what is expected.

Bug4: After investigating, this seems to be a bug on GSI's (Google) end. There have been several issues filed across different GH repos, which lead us to this conclusion. I'll provide a few links below. But from what we can tell, this might be a non-issue, other than the warning. The sign-in function proceeds as expected.
[COOP HTTP Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy)
[Issue on Vercel](https://github.com/vercel/next.js/discussions/51135)
[Issue on another Google sign in library](https://github.com/MomenSherif/react-oauth/issues/289)
[Google docs about COOP](https://developers.google.com/identity/gsi/web/guides/get-google-api-clientid#cross_origin_opener_policy)
@marcochavezf what are your thoughts on this? Is it possible we could set this configuration on the backend? None of the attempts to change configurations in the client seemed to help.

Bug11: This was a result of us use `Log.error`, which turns out doesn't exist. We switched to use a different method and to be more specific in the messaging.

Let us know if there's any followup questions or concerns!

cdanwards committed:
Update use of Log

marcochavezf commented:
There are some conflicts 😬

cdanwards committed:
Merge branch 'main' into new-third-party-signin

cdanwards committed:
Fixed merge conflicts from main

cdanwards commented:
@marcochavezf correct, it's not a blocker but something to be aware of.

marcochavezf commented:
Seems there are conflicts again, given the size of the PR I think it will be a bit hard to find a moment where there are no new conflicts 😅 

@situchan once the conflicts are resolved, let's do a final review to merge it asap, then we can deploy it to staging on Monday and test the full flows there. With all the changes addressed in this PR and the previous one, I think there are no risks of having another regression

cdanwards committed:
Merge branch 'main' into new-third-party-signin

situchan commented:
> @situchan once the conflicts are resolved, let's do a final review to merge it asap, then we can deploy it to staging on Monday and test the full flows there. With all the changes addressed in this PR and the previous one, I think there are no risks of having another regression

Sure, I will review again during weekend,

@cdanwards Also, please fix lint

cdanwards committed:
Fix lint error

cdanwards commented:
@situchan lint error fixed! Was an issue from the merge conflicts.

situchan commented:
Bug1,6: Isn't there any way to fix this in dev? Anyway, let's create follow-up GH for tracking. Otherwise anyone will report this as bug and claim bonus after fix. Same for Bug 4.

Bug2: I don't think `minHeight` change is the right solution. This will cause regression on small height devices. Please check diff on iOS

https://github.com/Expensify/App/assets/108292595/ee343744-fdd7-46cf-8edc-27070a16ea32

Bug3: We should still discuss offline behavior. Otherwise bug reporters will continue to report this as regression. i.e. https://github.com/Expensify/App/issues/23257, https://github.com/Expensify/App/issues/23265. Even marking this as deploy blocker.

Bug5: This is still reproducible to me on Mac Chrome. Please follow exact repro step as in my video. (it should be just localhost:8082 without any path)


https://github.com/Expensify/App/assets/108292595/4c7c3484-e95c-4495-ab08-9e4cabfa63f6

Bug11: please fix on android as well. `Log.error` still exists in android file


I think we should fix at least Bug 2, 5 and 11 before merge.
_Other bugs are fixed or not blocker._

cdanwards committed:
remove use of Log.error

cdanwards commented:
@situchan 

For Bug1 & Bug6 - We're unable to change this. @marcochavezf should I go ahead and create a bug for this and explain there that this is expected behavior in development?

As for Bug2, if we don't change the `minHeight` on that element (or on the parent component where the `minHeight` is set to [here](https://github.com/Expensify/App/blob/bebbaa2d1d4b72103cded70be1da755297821eba/src/pages/signin/SignInPageLayout/index.js#L124)) we would have to change how the component structure uses spacing to alternatively use margins and padding instead of `minHeight`. Because of the flexed nature of the components, the children will _always_ fill the space available to them, and that's what's happening when the new buttons appear. @marcochavezf any thoughts?

Bug4: I'll create a GH issue for this as well to track.

Bug5: I believe this is because the `CONFIG.ENVIRONMENT` is equal to `CONST.ENVIRONMENT.DEV`. [Reference](https://github.com/Expensify/App/blob/bebbaa2d1d4b72103cded70be1da755297821eba/src/components/DeeplinkWrapper/index.website.js#L28) To fix this you can set `CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.STAGING` above the line linked to fix this. The deeplink _does not_ work in the development environment because of that check. You have to manually override the `CONFIG` to test.

Apologies for Bug11. That one is fixed now.

situchan commented:
> Bug5: I believe this is because the `CONFIG.ENVIRONMENT` is equal to `CONST.ENVIRONMENT.DEV`. [Reference](https://github.com/Expensify/App/blob/bebbaa2d1d4b72103cded70be1da755297821eba/src/components/DeeplinkWrapper/index.website.js#L28) To fix this you can set `CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.STAGING` above the line linked to fix this. The deeplink _does not_ work in the development environment because of that check. You have to manually override the `CONFIG` to test.

This doesn't happen on main. dev env of course. I won't have reported if it also exists on main.

situchan commented:
> > This doesn't happen on main. dev env of course. I won't have reported if it also exists on main.
> 
> What doesn't happen on main?

Open desktop app popup shows. Please test yourself

cdanwards commented:
> Open desktop app popup shows. Please test yourself

I just tested `localhost:8082` on both `main` and `new-third-party-signin` and I did not see the prompt on either. I'm confused what the bug is here.

situchan commented:
> > Open desktop app popup shows. Please test yourself
> 
> I just tested `localhost:8082` on both `main` and `new-third-party-signin` and I did not see the prompt on either. I'm confused what the bug is here.

main:

https://github.com/Expensify/App/assets/108292595/82c12dec-0e57-49f9-b2ce-c0c6bf42c08a

situchan commented:
You should comment this line to test locally:

https://github.com/Expensify/App/blob/bebbaa2d1d4b72103cded70be1da755297821eba/src/components/DeeplinkWrapper/index.website.js#L28-L30

We can fix this as follow-up if needed to merge asap. But let's create GH for tracking.

cdanwards committed:
Added comment to clarify flow

cdanwards commented:
@situchan I think there was a bit of context lacking. With this PR, part of the design is that there is logic introduced where if the user is unauthenticated, certain screens don't present the deeplink prompt.

We immediately present the prompt to authenticated users, otherwise we check whether it is a screen where deeplinking is blocked. Currently that list of screens is `/`, `sign-in-with-google`, and `sign-in-with-apple`

Sorry, I think there was a bit of miscommunication, but I think we are both seeing behavior that is expected.

situchan commented on src/CONST.js:
```diff
@@ -425,6 +426,15 @@ const CONST = {
 
+    SIGN_IN_FORM_WIDTH: 300,
+
+    DEEPLINK_PROMPT_DENYLIST: [SCREENS.HOME, SCREENS.SIGN_IN_WITH_APPLE_DESKTOP, SCREENS.SIGN_IN_WITH_GOOGLE_DESKTOP],
```
Gotcha. I thought HOME is another link than root `/`

marcochavezf commented:
> For Bug1 & Bug6 - We're unable to change this. @marcochavezf should I go ahead and create a bug for this and explain there that this is expected behavior in development?

@situchan @cdanwards I think it would be better to hide the buttons in development mode because I agree that the community will be reporting and will create back-and-forth with the team that creates the GH issues and the internal team.   We could do something like we do for deep link to desktop: 

```js
// Third-party doesn't work out of the box in development mode. Read contributingGuides/APPLE_GOOGLE_SIGNIN.md for more instructions
if (CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.DEV) { 
     return; 
 } 
 ```

marcochavezf commented:
> As for Bug2, if we don't change the minHeight on that element (or on the parent component where the minHeight is set to [here](https://github.com/Expensify/App/blob/bebbaa2d1d4b72103cded70be1da755297821eba/src/pages/signin/SignInPageLayout/index.js#L124)) we would have to change how the component structure uses spacing to alternatively use margins and padding instead of minHeight. Because of the flexed nature of the components, the children will always fill the space available to them, and that's what's happening when the new buttons appear. @marcochavezf any thoughts?

Could it be solved by changing the minHeight with a media query only for large devices?

marcochavezf commented:
> Bug4: [web] console error when start google signin process

For bug 4, could that be caused by the localhost env? If that doesn't happen in staging/production that will be fine but I think we would only know until we deploy

cdanwards commented:
> Could it be solved by changing the minHeight with a media query only for large devices?

Change this to something like
` isSmallScreen ? minHeight: 45 : minHeight: 24`?

> For bug 4, could that be caused by the localhost env? If that doesn't happen in staging/production that will be fine but I think we would only know until we deploy

I tested this in an SSH tunnel, which mimics staging or production and unfortunately I continued to see this error.

cdanwards committed:
Prevent rendering third party sign in buttons in development

cdanwards commented:
@marcochavezf @situchan I updated the `LoginForm` to not render the third party signin button in development. I also updated the accompanying contributing guide with info about removing that check.

situchan commented:
>  minHeight: isSmallScreen ? 24 : 45

This is not perfect solution but looks better.

I think this one is the last blocker.
@marcochavezf shall we go ahead with this change?

marcochavezf commented:
> > minHeight: isSmallScreen ? 24 : 45
> 
> This is not perfect solution but looks better.
> 
> I think this one is the last blocker. @marcochavezf shall we go ahead with this change?

I don't see a problem with this, we use `isSmallScreenWidth` in the code base:

```js
    const {isSmallScreenWidth} = useWindowDimensions();
```

I think a cleaner way is to define the values as constants and use them along with `isSmallScreenWidth`, something like:

```js
isSmallScreenWidth ? styles.signInPageContentTopSpacerSmallScreens : styles.signInPageContentTopSpacer
```

marcochavezf commented:
> > For bug 4, could that be caused by the localhost env? If that doesn't happen in staging/production that will be fine but I think we would only know until we deploy
> 
> I tested this in an SSH tunnel, which mimics staging or production and unfortunately I continued to see this error.

Would it be related to CSP? I remember we had an issue with the Apple cdn with the Apple sign-in, I wonder if we should add something similar

lindboe commented:
> Would it be related to CSP? I remember we had an issue with the Apple cdn with the Apple sign-in, I wonder if we should add something similar

No, it's an overly-eager verification error on Google's end. [What Google says in their docs](https://developers.google.com/identity/gsi/web/guides/get-google-api-clientid#cross_origin_opener_policy) is to ensure you don't use `same-origin` as the value for the `Cross-Origin-Opener-Policy` header, and to instead use `same-origin-allow-popups`. The Expensify app isn't doing this, it isn't setting it at all, and the default value (`unsafe-none`) is [more permissive than all of the options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy#directives). So the client library is erroneously warning this won't work because it's not set to `same-origin-allow-popups`, but `unsafe-none` will allow everything `same-origin-allow-popups` does and more.

I've looked for how to try to report this bug to Google but have not found anything.

We _could_ add the header and change it to `same-origin-allow-popups` like we did with CSP, and that would fix the warning, but it could potentially break something else.

By the way, we do need to get CSP headers updated for Google according to [these steps,](https://developers.google.com/identity/gsi/web/guides/get-google-api-clientid#content_security_policy) not sure if @cdanwards already mentioned that?

marcochavezf commented:
> By the way, we do need to get CSP headers updated for Google according to [these steps,](https://developers.google.com/identity/gsi/web/guides/get-google-api-clientid#content_security_policy) not sure if @cdanwards already mentioned that?

Should we add all the directives? 

```
Content-Security-Policy-Report-Only: script-src https://accounts.google.com/gsi/client; frame-src https://accounts.google.com/gsi/; connect-src https://accounts.google.com/gsi/;
```

lindboe commented:
We only need to add values for the directives if they're already being set; however from looking at new.expensify.com it looks like we're using all of those, so yes.

cdanwards commented:
> > > minHeight: isSmallScreen ? 24 : 45
> > 
> > 
> > This is not perfect solution but looks better.
> > I think this one is the last blocker. @marcochavezf shall we go ahead with this change?
> 
> I don't see a problem with this, we use `isSmallScreenWidth` in the code base:
> 
> ```js
>     const {isSmallScreenWidth} = useWindowDimensions();
> ```
> 
> I think a cleaner way is to define the values as constants and use them along with `isSmallScreenWidth`, something like:
> 
> ```js
> isSmallScreenWidth ? styles.signInPageContentTopSpacerSmallScreens : styles.signInPageContentTopSpacer
> ```

Yeah this looks good! I'll make the change. I was pseudocoding to get the point across 🙂 

I'll get these changes up.

marcochavezf commented:
> We only need to add values for the directives if they're already being set; however from looking at new.expensify.com it looks like we're using all of those, so yes.

Cool, thanks for confirming. I requested to add the directives to our CSP

cdanwards committed:
Make the page content top spacer minHeight conditional

cdanwards commented:
@marcochavezf that last change is in!

marcochavezf commented:
The CSP changes were deployed to staging and production, so the new CSP will be ready once the PR is deployed. @situchan would you like to do a final review? 

@cdanwards I was thinking if it would be possible to set up a ngrok endpoint for @situchan to test locally and complete the PR checklist?

situchan commented:
I think we can consider CSP issue not blocker for merging this PR.

cdanwards commented:
Yeah! @situchan it's pretty easy to get your authToken and agent up and running. Here's the explanation on their site: https://dashboard.ngrok.com/get-started/setup - Once you login they list through the steps of setting it up and running the agent.

situchan commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

cdanwards commented:
@marcochavezf are there any other checks we need to do to get this through?

Also, just to clarify. I'm going to create an issue preemptively for the Bug4 (COOP error likely from Google's GSI), were there any other errors that should have issues created to track? I know we solved the white background issue on dev by hiding the buttons on dev.

marcochavezf commented:
We're just tracking the issue [here](https://github.com/Expensify/App/issues/7079), we can link the bug4 there

situchan commented:
@marcochavezf We should test web based platforms after deploy to staging, right?

cdanwards commented:
@marcochavezf just to make sure I'm on the same page. Link the conversation about bug4 in the tracking issue, or create a new issue about bug4 and link that in the tracking issue?

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

marcochavezf commented:
Ah sorry for the confusion @cdanwards, let's create a new issue for bug4 and link it to the tracking issue please

cdanwards commented:
@marcochavezf no worries! Just wanted to make sure I understood. I'll get that issue created!

lindboe commented:
It was noted Google Sign-In isn't working on ad hoc builds: https://expensify.slack.com/archives/C01GTK53T8Q/p1692302447001879

It looks like logging isn't quite getting all the details we'd want. I'm creating a PR to quickly address that.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5931749076) to staging by https://github.com/marcochavezf in version: 1.3.56-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

dangrous commented:
Hi all! See the above deploy blocker - https://github.com/Expensify/App/issues/25768 - Can we look into why that Google sign in button jumps on desktop Safari? Thanks!

cdanwards commented:
@dangrous Hey! I'm taking a look at this! We mainly tested on Chrome and Brave browsers during development and I haven't been able to reproduce there (We were informed that Firefox wasn't supported). Is Safari one of the supported browsers? It definitely seems like an issue of the iframe rendering and possible re-rendering the button and the styles being updated on a millisecond delay.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5958899174) to production by https://github.com/roryabraham in version: 1.3.56-24 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

napster125 commented on src/components/DeeplinkWrapper/index.website.js:
```diff
@@ -1,45 +1,95 @@
+            setHasShownPrompt(true);
+        } else {
+            // Navigation state is not set up yet, we're unsure if we should show the deep link prompt or not
+            if (currentScreen === undefined || isAuthenticated === false) {
```
In this context, why would we use `isAuthenticated === false`?

ArekChr replied:
cc: @cdanwards 

https://github.com/Expensify/App/issues/25790

cdanwards replied:
@napster125 We don't want to show the deeplink prompt to unauthenticated users.

cdanwards replied:
There's discussion of it within this feature's design document. Trying to find if we included that conversation in the contributing guide.

cdanwards replied:
Following up, here are the slack threads where this was discussed:
https://expensify.slack.com/archives/C01GTK53T8Q/p1683070404415199
https://expensify.slack.com/archives/C01GTK53T8Q/p1689886614735019

alitoshmatov commented on src/components/DeeplinkWrapper/index.website.js:
```diff
@@ -1,45 +1,95 @@
-
-        if (!isMacOSWeb() || isUnsupportedDeeplinkRoute || CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.DEV) {
+        // Making a few checks to exit early before checking authentication status
+        if (!isMacOSWeb() || isUnsupportedDeeplinkRoute || CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.DEV || hasShownPrompt) {
             return;
         }
+        // We want to show the prompt immediately if the user is already authenticated.
+        // Otherwise, we want to wait until the navigation state is set up
+        // and we know the user is on a screen that supports deeplinks.
+        if (isAuthenticated) {
+            promptToOpenInDesktopApp();
+            setHasShownPrompt(true);
+        } else {
+            // Navigation state is not set up yet, we're unsure if we should show the deep link prompt or not
+            if (currentScreen === undefined || isAuthenticated === false) {
```
These conditions did not consider opening magic link on logged in user, and caused this issue https://github.com/Expensify/App/issues/32795
